### DISCUSSION TITLE
`422 Unprocessable Entity` for `GET`

### DISCUSSION THREAD

**Discussion by baniasbaabe** [fastapi/fastapi#10782] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import uvicorn
from fastapi import FastAPI
from pydantic import BaseModel

app = FastAPI(debug=True)
class GetTemplates(BaseModel):
    notionKey: str
    databaseId: str

@app.get(
    "/get_templates",
)
async def get_templates(data: GetTemplates):
    print(data.notionKey, data.databaseId)


if __name__ == "__main__":
    uvicorn.run(app, host="0.0.0.0", port=8000)
```

### Description

Whenever I try to request the endpoint with `axios` I get an 422 Error. 

```javascript
axios.get('http://localhost:8000/get_templates',
      {params: { notionKey: notionKey, databaseId: databaseId }, headers: {
        Accept: 'application/json',
        ContentType: "application/json"
    }})
``` 

But, whenever I change the parameters of my `GET` methods to

```python
async def get_templates(notionKey: str, databaseId: str):
```

which is weird, since it's basically the same. For my `POST` methods, I can use pydantic models without problems.

### Operating System

Linux

### FastAPI Version

0.104.1

### Pydantic Version

2.5.2

### Python Version

3.8.7

**Comment by mntolia** [comment, marked as answer]
Pydantic models are meant to be used for the request body.

You shouldn't use a GET request to send sensitive information. The URL is not encrypted when TLS is used.

Either way in your axios request the 'ContentType' header should be 'Content-Type'. You forgot the hyphen.

**Reply by baniasbaabe** [reply]
Thanks for the clarification! Just curious to know how to make a „secure“ GET request with sensitive information with FastAPI

**Reply by mntolia** [reply]
You can tokenize or encrypt the data manually. Eitherway its a bad practice whether you use FastAPI or any other framework.

**Reply by vincentfenet** [reply]
With https://foo.example.com/bar, only the hostname (foo.example.com) could be leaked, see [SNI](https://en.wikipedia.org/wiki/Server_Name_Indication)
